Core/include/Acts/EventData/TrackStateProxy.hpp

```diff
@@ -386,8 +386,8 @@ class TrackStateProxy {
   /// This overloaded is only enabled if not read-only, and returns a mutable
   /// reference.
   /// @return Mutable reference to the pathlength.
-  template <bool RO = ReadOnly, typename = std::enable_if_t<!RO>>
-  double& pathLength() {
+  template <bool RO = ReadOnly>
+  requires(!RO) double& pathLength() {
```

Comment:
Seems like I missed these though 😅 
```suggestion
  requires(!ReadOnly) double& pathLength() {
```

Reply:
Aside from that, in the rest of the changes I've made so far, I've put the `requires` clause behind the function name. Maybe we can keep that consistent?